tests/test_read_backed_experimental.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+from importlib.util import find_spec
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+from scipy import sparse
+
+from anndata._core.anndata import AnnData
+from anndata.experimental import read_backed
+from anndata.tests.helpers import (
+    AccessTrackingStore,
+    as_dense_dask_array,
+    assert_equal,
+    gen_adata,
+    gen_typed_df,
+)
+
+
+@pytest.fixture(
+    params=[sparse.csr_matrix, sparse.csc_matrix, np.array, as_dense_dask_array],
+    ids=["scipy-csr", "scipy-csc", "np-array", "dask_array"],
+    scope="session",
+)
+def mtx_format(request):
+    return request.param
+
+
+@pytest.fixture(params=["zarr", "h5ad"], scope="session")
+def dskfmt(request):
+    return request.param
+
+
+@pytest.fixture(scope="session")
+def adata_remote_orig(
+    tmp_path_factory, dskfmt: str, mtx_format
+) -> tuple[AnnData, AnnData]:
+    orig_path = Path(tmp_path_factory.mktemp(f"orig.{dskfmt}"))
+    orig = gen_adata((1000, 1000), mtx_format)
+    orig.write_zarr(orig_path)
+    remote = read_backed(orig_path)
+    return remote, orig
```

Comment:
This is session-scoped fixture returning a backed AnnData object.

What’s the idea here? Tests shouldn’t be interdependent, so I assume the tests want to uphold the invariant of not mutating these objects. Can guarantee somehow that we uphold this invariant? E.g. opening readonly or making this function-scoped, or so?

Reply:
So the idea here is that `read_backed` is read-only actually (if you look, all `mode="r"` there)